### ISSUE TITLE
service/loki missing following "monolithic Helm chart" install.

### ISSUE THREAD

**Issue by javydekoning** [grafana/loki#13045]
**Describe the bug**
Following the docs for Monolithic helm install, no loki service is created (3100)

**To Reproduce**

Values: 

```yaml
loki:
  mode: SingleBinary
  loki:
    auth_enabled: false
    commonConfig:
      replication_factor: 1
    storage:
      type: 'filesystem'
    schemaConfig:
      configs:
      - from: "2024-01-01"
        store: tsdb
        index:
          prefix: loki_index_
          period: 24h
        object_store: filesystem # we're storing on filesystem so there's no real persistence here.
        schema: v13
  singleBinary:
    replicas: 1
  read:
    replicas: 0
  backend:
    replicas: 0
  write:
    replicas: 0
```

Run `helm template . --values ./values.yaml | k apply -f -`

And no service exists:

```
loki-canary                       ClusterIP   172.20.231.211   <none>        3500/TCP             6m55s
loki-memberlist                   ClusterIP   None             <none>        7946/TCP             6m55s
release-name-loki-chunks-cache    ClusterIP   None             <none>        11211/TCP,9150/TCP   6m55s
release-name-loki-gateway         ClusterIP   172.20.174.145   <none>        80/TCP               6m55s
release-name-loki-results-cache   ClusterIP   None             <none>        11211/TCP,9150/TCP   6m55s
```

**Expected behavior**
Loki service created TCP 3100

**Environment:**
k3s single node

**Comment by dk03051996** [comment]
hi @javydekoning , if you are looking for a service named loki with port as 3100 so fyi, it has been removed long back and you can use gateway service now to configure with grafana.

**Comment by javydekoning** [comment]
> hi @javydekoning , if you are looking for a service named loki with port as 3100 so fyi, it has been removed long back and you can use gateway service now to configure with grafana.

Doens't seem to work for me. From grafana logs:

```
938295Z level=error msg="Error received from Loki" duration=2.978292ms stage=databaseRequest statusCode=502 contentLength=157 start=1970-01-01T00:00:01Z end=1970-01-01T00:00:04Z step=1s query=vector(1)+vector(1) queryType=instant direction=backward maxLines=0 supportingQueryType=none lokiHost=loki-gateway:80 lokiPath=/loki/api/v1/query status=error error="<html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body>\r\n<center><h1>502 Bad Gateway</h1></center>\r\n<hr><center>nginx/1.24.0</center>\r\n</body>\r\n</html>\r\n" statusSource=downstream
logger=tsdb.loki endpoint=CheckHealth t=2024-05-28T13:41:33.641031352Z level=error msg="Loki health check failed" error="error from loki: <html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body>\r\n<center><h1>502 Bad Gateway</h1></center>\r\n<hr><center>nginx/1.24.0</center>\r\n</body>\r\n</html>\r\n"
logger=context userId=1 orgId=1 uname=admin t=2024-05-28T13:41:33.641138512Z level=info msg="Request Completed" method=GET path=/api/datasources/uid/P8E80F9AEF21F6940/health status=400 remote_addr=10.42.0.1 time_ms=9 duration=9.4896ms size=106 referer=https://grafana.<REMOVED>.com/connections/datasources/edit/P8E80F9AEF21F6940 handler=/api/datasources/uid/:uid/health status_source=server
```

Loki gateway logs:

```
2024/05/28 13:42:30 [error] 8#8: *4508 loki-query-frontend.monitoring.svc.cluster.local could not be resolved (3: Host not found), client: 10.42.0.110, server: , request: "GET /loki/api/v1/query_range?start=1716883918649835947&end=1716903247052277174&query=%7Bstream%3D%22stdout%22%2Cpod%3D%22loki-canary-f8v9q%22%7D+&limit=1000 HTTP/1.1", host: "loki-gateway.monitoring.svc.cluster.local.:80"
10.42.0.110 - - [28/May/2024:13:42:30 +0000]  502 "GET /loki/api/v1/query_range?start=1716883918649835947&end=1716903247052277174&query=%7Bstream%3D%22stdout%22%2Cpod%3D%22loki-canary-f8v9q%22%7D+&limit=1000 HTTP/1.1" 157 "-" "loki-canary/3.0.0" "-"
```

Loki-query-frontend service does not exist either with single binary deployment. Not sure how to debug this?

**Comment by javydekoning** [comment]
Yes, [docs are updated](https://grafana.com/docs/loki/latest/setup/install/helm/install-monolithic/). You should use `deploymentMode: SingleBinary` and NOT combine it with `gateway.enabled: true`. Then it'll be available on `loki:3100`
